image/dng.cpp

```diff
@@ -36,21 +36,23 @@ struct BayerFormat
 
 static const std::map<PixelFormat, BayerFormat> bayer_formats =
 {
-	{ formats::SRGGB10_CSI2P, { "RGGB-10", 10, TIFF_RGGB } },
-	{ formats::SGRBG10_CSI2P, { "GRBG-10", 10, TIFF_GRBG } },
-	{ formats::SBGGR10_CSI2P, { "BGGR-10", 10, TIFF_BGGR } },
-	{ formats::R10_CSI2P,     { "BGGR-10", 10, TIFF_BGGR } },
-	{ formats::SGBRG10_CSI2P, { "GBRG-10", 10, TIFF_GBRG } },
-	{ formats::SRGGB12_CSI2P, { "RGGB-12", 12, TIFF_RGGB } },
-	{ formats::SGRBG12_CSI2P, { "GRBG-12", 12, TIFF_GRBG } },
-	{ formats::SBGGR12_CSI2P, { "BGGR-12", 12, TIFF_BGGR } },
-	{ formats::SGBRG12_CSI2P, { "GBRG-12", 12, TIFF_GBRG } },
-	{ formats::SRGGB16,       { "RGGB-16", 16, TIFF_RGGB } },
-	{ formats::SGRBG16,       { "GRBG-16", 16, TIFF_GRBG } },
-	{ formats::SBGGR16,       { "BGGR-16", 16, TIFF_BGGR } },
-	{ formats::SGBRG16,       { "GBRG-16", 16, TIFF_GBRG } },
+    { formats::SRGGB10_CSI2P, { "RGGB-10", 10, TIFF_RGGB } },
+    { formats::SGRBG10_CSI2P, { "GRBG-10", 10, TIFF_GRBG } },
+    { formats::SBGGR10_CSI2P, { "BGGR-10", 10, TIFF_BGGR } },
+    { formats::R10_CSI2P,     { "BGGR-10", 10, TIFF_BGGR } },
+    { formats::SGBRG10_CSI2P, { "GBRG-10", 10, TIFF_GBRG } },
+    { formats::SRGGB12_CSI2P, { "RGGB-12", 12, TIFF_RGGB } },
+    { formats::SGRBG12_CSI2P, { "GRBG-12", 12, TIFF_GRBG } },
+    { formats::SBGGR12_CSI2P, { "BGGR-12", 12, TIFF_BGGR } },
+    { formats::SGBRG12_CSI2P, { "GBRG-12", 12, TIFF_GBRG } },
+    { formats::SRGGB16,       { "RGGB-16", 16, TIFF_RGGB } },
+    { formats::SGRBG16,       { "GRBG-16", 16, TIFF_GRBG } },
+    { formats::SBGGR16,       { "BGGR-16", 16, TIFF_BGGR } },
+    { formats::SGBRG16,       { "GBRG-16", 16, TIFF_GBRG } },
+    { formats::MONO8,         { "MONO-8",  8,  TIFF_MONO } }  // New entry for monochromatic
```

Comment:
Quite possible, I'm not familiar. However, I desperately need the function for a science project, since I need to have .raw images. What would you suggest sir?

Reply:
Which sensor are you using for your project?  If it is one of the official Raspberry Pi cameras, RAW files should be fully supported.

Reply:
I'm using an Arducam OV9281. Should I go ahead and purchase an Official camera?

Reply:
The OV9281 is a 10-bit sensor, so I expect it ought to work with the recent changes to add 10-bit mono support.  This is the commit in question: https://github.com/raspberrypi/libcamera-apps/commit/d932d3132a2414b976a08373ff8dc5e83f568ef5.  Note that this change has not made it's way into a release yet, so you will have to manually build libcamera-apps to test it out for now.